org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacBindingResolver.java

```diff
@@ -108,6 +108,7 @@ public class JavacBindingResolver extends BindingResolver {
 	public final IJavaProject javaProject;
 	private JavacConverter converter;
 	boolean isRecoveringBindings = false;
+	boolean hasResolved = false;
```

Comment:
I'm guessing at least one but not all of the symbols are populated, which prevents analyze from ever being called here. The root cause of the issue Gayan pinged me on was that some AST nodes (all the symbols in the class I was working with) didn't have symbols or types and I was expecting them to.

Reply:
I could switch it to check if all the javac nodes have symbols. I did made that change and ran the test suite, and it seems that sometimes its the case that all the javac nodes have symbols (because analysis has been run) but `this.symbolToDeclaration` hasn't been filled out yet.